src/main.rs

```diff
@@ -20,6 +20,54 @@ use app::*;
 use assets::*;
 use ui::input::*;
 
+#[cfg(target_family = "unix")]
+use winit::platform::unix::*;
+
+#[cfg(target_family = "unix")]
+fn winit_argb_from_skia_color(color: skia_safe::Color) -> ARGBColor {
+    ARGBColor {
+        a: color.a(),
+        r: color.r(),
+        g: color.g(),
+        b: color.b(),
+    }
+}
+
+#[cfg(target_family = "unix")]
+impl Theme for assets::ColorScheme {
+    fn element_color(&self, element: Element, window_active: bool) -> ARGBColor {
+        match element {
+            Element::Bar => winit_argb_from_skia_color(self.text_field.fill),
+            Element::Separator => winit_argb_from_skia_color(self.text_field.text_hint),
+            Element::Text => winit_argb_from_skia_color(self.text_field.text),
+        }
+    }
+
+    fn button_color(&self, button: Button, state: ButtonState, foreground: bool, _window_active: bool) -> ARGBColor {
+        let color: ARGBColor;
+
+        match button {
+            Button::Close => color = winit_argb_from_skia_color(self.error),
+            Button::Maximize => color = winit_argb_from_skia_color(self.text),
+            Button::Minimize => color = winit_argb_from_skia_color(self.text),
+        }
```

Comment:
You can turn this into a match expression to avoid the repeated assignment to color, like `let color = match button { … };`.